tests/test_two_moons/test_two_moons.py

```diff
@@ -11,14 +11,38 @@ def test_compile(approximator, random_samples, jit_compile):
 
 
 @pytest.mark.parametrize("jit_compile", [False, True])
-def test_fit(approximator, train_dataset, validation_dataset, jit_compile):
-    # TODO: verify the model learns something by comparing a metric before and after training
-    approximator.compile(jit_compile=jit_compile)
-    approximator.fit(
+def test_fit(approximator, train_dataset, validation_dataset, test_dataset, jit_compile):
+    # TODO: Refactor to use approximator.sample() when implemented (instead of calling the inference network directly)
+
+    approximator.compile(jit_compile=jit_compile, loss=keras.losses.KLDivergence())
+    inf_vars = approximator.configurator.configure_inference_variables(test_dataset.data)
+    inf_conds = approximator.configurator.configure_inference_conditions(test_dataset.data)
+    y = test_dataset.data["x"]
```

Comment:
relies on the internal structure of the dataset fixture, which is not good. Use `test_batch = test_dataset[0]; observables = test_batch["x"]` instead.